proposals/2020-07-01_aws-kms-keyring-redesign/change.md

```diff
@@ -0,0 +1,327 @@
+
+## Affected Implementations
+
+This serves as a reference for all implementations that this change affects.
+
+| Language   | Implementation                                                                        |
+| ---------- | ------------------------------------------------------------------------------------- |
+| C          | [aws-encryption-sdk-c](https://github.com/aws/aws-encryption-sdk-c)                   |
+| Javascript | [aws-encryption-sdk-javascript](https://github.com/aws/aws-encryption-sdk-javascript) |
+
+This change additionally affects the unreleased AWS KMS keyring implementations for Java and Python.
+
+## Definitions
+
+### Conventions used in this document
+
+The key words
+"MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+"SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL"
+in this document are to be interpreted as described in
+[RFC 2119](https://tools.ietf.org/html/rfc2119).
+
+## Summary
+
+The AWS Encryption SDK’s (ESDK’s) [AWS KMS Keyring](https://github.com/awslabs/aws-encryption-sdk-specification/blob/dbc17f93100667e28dc54e64d05a625db3e5bac2/framework/kms-keyring.md) has become increasingly complex
+as it [supports various customer use cases](background.md).
+We are separating the functionality of the AWS KMS keyring into multiple smaller-scoped keyrings.
+Each new keyring maintains a smaller API surface area.
+
+By simplifying the AWS KMS keyring, we establish the basic building blocks for interacting with AWS KMS.
+We can support additional customer use cases by writing operations to configure a multi-keyring with these smaller-scoped keyrings.
+In the future, new features will be supported without refactoring code or updating customer-facing APIs.
+Advanced customers can also follow a similar process for more complex customization.
+This lowers the likelihood that advanced customers fork ESDK code.
+
+## Out of Scope
+
+1. Any customer interactions with the ESDK that are unrelated to AWS KMS
+2. Keyrings other than the AWS KMS keyring
+   1. We rely on the functionality of the [multi-keyring](https://github.com/awslabs/aws-encryption-sdk-specification/blob/master/framework/multi-keyring.md),
+      but the multi-keyring is out of scope/unchanged
+3. [AWS KMS asymmetric CMKs](https://docs.aws.amazon.com/kms/latest/developerguide/symm-asymm-concepts.html#asymmetric-cmks)
+
+## Motivation
+
+1. **Move logic out of the AWS KMS keyring** so a single keyring does not handle
+   multiple AWS KMS [CMKs](https://docs.aws.amazon.com/kms/latest/developerguide/concepts.html#master_keys)
+   ([key names](https://github.com/awslabs/aws-encryption-sdk-specification/blob/dbc17f93100667e28dc54e64d05a625db3e5bac2/framework/kms-keyring.md#key-names)),
+   multiple AWS regions,
+   or complex functionality (
+   [discovery](https://github.com/awslabs/aws-encryption-sdk-specification/blob/dbc17f93100667e28dc54e64d05a625db3e5bac2/framework/kms-keyring.md#is-discovery),
```

Comment:
Yup! This still interprets properly! Updating here and in background.md. 

For this change, and any suggestions like this below, I'll update them myself since "Add suggestion to batch" doesn't appear to work. Also, in some cases, I have to update both documents so doing it myself ensures I don't forget any changes.